src/ListTabulator/index.ts

```diff
@@ -445,274 +489,365 @@ export default class ListTabulator {
+    }
 
+    this.focusItem(itemEl);
+  }
 
   /**
-   * Reduce indentation for current item
-   *
-   * @param {KeyboardEvent} event - keydown
-   * @returns {void}
+   * Method that is used for merging current item with previous one
+   * Content of the current item would be appended to the previous item
+   * Current item children would not change nesting level
+   * @param currentItem - current item html element
    */
-  shiftTab(event: KeyboardEvent): void {
+  mergeItemWithPrevious(item: ItemElement): void {
+    const previousItem = item.previousElementSibling;
+
+    const currentItemParentNode = item.parentNode;
+
     /**
-     * Prevent editor.js behaviour
+     * Check that parent node of the current element exists
      */
-    event.stopPropagation();
+    if (currentItemParentNode === null) {
+      return;
+    }
+    if (!isHtmlElement(currentItemParentNode)) {
+      return;
+    }
+
+    const parentItem = currentItemParentNode.closest<ItemElement>(`.${DefaultListCssClasses.item}`);
 
     /**
-     * Prevent browser tab behaviour
+     * Check that current item has any previous siblings to be merged with
      */
-    event.preventDefault();
+    if (!previousItem && !parentItem) {
+      return;
+    }
 
     /**
-     * Move item from current list to parent list
+     * Make sure previousItem is an HTMLElement
      */
-    this.unshiftItem();
-  }
-
-
-  /**
-   * Decrease indentation of the current item
-   *
-   * @returns {void}
-   */
-  unshiftItem(): void {
-    const currentItem = this.currentItem;
-    if (!currentItem) {
+    if (previousItem && !isHtmlElement(previousItem)) {
       return;
     }
-    if (!currentItem.parentNode) {
-      return;
+
+    /**
+     * Lets compute the item which will be merged with current item text
+     */
+    let targetItem: ItemElement | null;
+
+    /**
+     * If there is a previous item then we get a deepest item in its sublists
+     *
+     * Otherwise we will use the parent item
+     */
+    if (previousItem) {
+      /**
+       * Get list of all levels children of the previous item
+       */
+      const childrenOfPreviousItem = previousItem.querySelectorAll<ItemElement>(`.${DefaultListCssClasses.item}`);
+
+      /**
+       * Target item would be deepest child of the previous item or previous item itself
+       */
+      targetItem = childrenOfPreviousItem[childrenOfPreviousItem.length - 1] || previousItem;
+    } else {
+      targetItem = parentItem;
     }
-    if (!isHtmlElement(currentItem.parentNode)) {
+
+    /**
+     * Get current item content
+     */
+    const currentItemContent = this.renderer.getItemContent(item);
+
+    /**
+     * Get the target item content element
+     */
+    if (!targetItem) {
       return;
     }
 
-    const parentItem = currentItem.parentNode.closest(`.${DefaultListCssClasses.item}`);
+    /**
+     * Get target item content element
+     */
+    const targetItemContentElement = getItemContentElement(targetItem);
```

Comment:
what is the difference between `this.renderer.getItemContent()`?

Reply:
getItemContent returns content as a `ListData['content']`, but method `getItemContentElement` returns HTMLElement